src/playground-code-editor.ts

```diff
@@ -633,6 +639,22 @@ export class PlaygroundCodeEditor extends LitElement {
       }
     );
 
+    // As CodeMirror doesn't let us directly query if the completion hints are shown,
+    // we want to have our own local state following the completions menu state.
+    CodeMirror.on(hints, 'shown', () => {
+      // Delay updating the status by a frame so that key listeners still have
+      // access to the correct state for the current situation.
+      window.requestAnimationFrame(() => {
```

Comment:
Does showing the hints require a `requestAnimationFrame`?

Reply:
Yeah both of them require it since the 'close' required it due to the comment. The start required it as these both get triggered when completions update (so on every keypress), and if only the close has it, it will always remain closed as the open status will race ahead of the close unless it has a requestanimationframe